guides/keras_cv/segment_anything_in_keras_cv.py

```diff
@@ -0,0 +1,500 @@
+"""
+
+"""
+Let's also visualize the other masks the model has predicted.
+"""
+
+fig, ax = plt.subplots(1, 3, figsize=(20, 60))
+masks, scores = outputs["masks"][0][1:], outputs["iou_pred"][0][1:]
+for i, (mask, score) in enumerate(zip(masks, scores)):
+    mask = inference_resizing(mask[..., None], pad=False)[..., 0]
+    mask, score = map(ops.convert_to_numpy, (mask, score))
+    mask = 1 * (mask > 0.0)
+    ax[i].imshow(ops.convert_to_numpy(image) / 255.0)
+    show_mask(mask, ax[i])
+    show_points(input_point, input_label, ax[i])
+    ax[i].set_title(f"Mask {i+1}, Score: {score:.3f}", fontsize=12)
+    ax[i].axis("off")
+plt.show()
+
+"""
+Nice! SAM was able to capture the ambiguity of our point prompt and also returned other
+possible segmentation masks.
+"""
+
+"""
+## Box Prompts
+
+Now, let's see how we can prompt the model using boxes. The box is specified using two
+points, the top-left corner and the bottom-right corner of the bounding box in xyxy
+format. Let's prompt the model using a bounding box around the left front tyre of the
+truck.
+"""
+
+# Let's specify the box
+input_box = np.array([[240, 340], [400, 500]])
+
+outputs = model.predict(
+    {"images": image[np.newaxis, ...], "boxes": input_box[np.newaxis, np.newaxis, ...]}
+)
+mask = inference_resizing(outputs["masks"][0][0][..., None], pad=False)[..., 0]
+mask = ops.convert_to_numpy(mask) > 0.0
+
+plt.figure(figsize=(10, 10))
+plt.imshow(ops.convert_to_numpy(image) / 255.0)
+show_mask(mask, plt.gca())
+show_box(input_box, plt.gca())
+plt.axis("off")
+plt.show()
+
+"""
+Boom! The model perfectly segments out the left front tyre in our bounding box.
+
+## Combining prompts
+
+To get the true potential of the model out, let's combine box and point prompts and see
+what the model does.
+"""
+
+# Let's specify the box
+input_box = np.array([[240, 340], [400, 500]])
+# Let's specify the point and mark it background
+input_point = np.array([[325, 425]])
+input_label = np.array([0])
+
+outputs = model.predict(
+    {
+        "images": image[np.newaxis, ...],
+        "points": input_point[np.newaxis, ...],
+        "labels": input_label[np.newaxis, ...],
+        "boxes": input_box[np.newaxis, np.newaxis, ...],
+    }
+)
+mask = inference_resizing(outputs["masks"][0][0][..., None], pad=False)[..., 0]
+mask = ops.convert_to_numpy(mask) > 0.0
+
+plt.figure(figsize=(10, 10))
+plt.imshow(ops.convert_to_numpy(image) / 255.0)
+show_mask(mask, plt.gca())
+show_box(input_box, plt.gca())
+show_points(input_point, input_label, plt.gca())
+plt.axis("off")
+plt.show()
+
+"""
+Voila! The model understood that the object we wanted to exclude from our mask was the
+rim of the tyre.
+
+## Text prompts
+
+Finally, let's see how text prompts can be used along with KerasCV's
+`SegmentAnythingModel`.
+
+For this demo, we will use the [offical grounding dino
+model](https://github.com/IDEA-Research/GroundingDINO). Grounding DINO is a model that
```

Comment:
For markdown links to render properly, they need to be on a single line, unbroken. The `markdown` library doesn't support links like the above (GitHub and the like presumably use a private fork that paves over these issues)

Reply:
Done, thanks!